federation/cmd/federated-apiserver/app/server.go

```diff
@@ -390,9 +240,8 @@ func Run(s *options.APIServer) error {
 
 	var serviceAccountGetter serviceaccount.ServiceAccountTokenGetter
 	if s.ServiceAccountLookup {
-		// If we need to look up service accounts and tokens,
-		// go directly to etcd to avoid recursive auth insanity
-		serviceAccountGetter = serviceaccountcontroller.NewGetterFromStorageInterface(etcdStorage)
+		// TODO: Maybe do not expose this option in genericapiserver, if most servers do not need it?
```

Comment:
> Should serviceAccountLookup be set to true for ubernetes? I think we all agree on "no". Ubernetes server does not serve tokens and service accounts.

Does ubernetes accept service account token auth? If so, the option should exist. I'd be fine with making the default match the default for the apiserver.

> Should serviceAccountLookup be set to true for core kubernetes server? Seems like we want to set it to true, but we do not currently. I can file a separate issue for that and fix it. I dont know enough about it to understand all the implications of that change.

I'd like to, @erictune wondered about possible performance impact originally, which is why it defaulted off. A separate issue would be good to investigate turning it on (maybe with a cache if there is performance impact).

> Should genericapiserver provide the ability? I guess if we decide to set it to true in core kubernetes server, then we can let that option be in genericapiserver. My proposal to delete it was based on the current state, where we do not set it anywhere.

Defaulting to false is not the same as not setting anywhere... it's an exposed flag, so it could be turned on in people's deployments.

> @liggitt and @deads2k do you guys set is to true in openshift server?

absolutely